### ISSUE TITLE
Rule proposal: prohibited-strings

### ISSUE THREAD

**Issue by gajus** [sindresorhus/eslint-plugin-unicorn#1553]
To prevent certain string from being used in the project, e.g. In Relay project, any use of `%future added value` string is indication that engineer misunderstood how to add value guards, e.g.

```js
if (input === '%future added value') {
  // This check should never exist.
}
```

## Fail

```js
var replace = `%future added value`;
```

## Pass

```js
const replace = 'me';
```

## Previous art

https://www.npmjs.com/package/eslint-plugin-prohibited-strings

**Comment by sindresorhus** [comment]
I think the rule name should be `no-restricted-strings` to follow ESLint convention:

<img width="235" alt="Screen Shot 2021-12-17 at 13 45 31" src="https://user-images.githubusercontent.com/170270/146546647-31e92555-0d12-4276-a0cb-a6380d27da90.png">

**Comment by sindresorhus** [comment]
I should also be possible to specify a regex instead of an exact string.

And optionally a message to show.

**Comment by sindresorhus** [comment]
@fisker 👍 👎 ?

**Comment by fisker** [comment]
I don't have need for it, but I won't object.
